frontend/packages/console-dynamic-plugin-sdk/src/extensions/user-preferences.ts

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { Extension, ExtensionDeclaration, CodeRef } from '../types';
+
+export enum UserPreferenceFieldType {
+  dropdown = 'dropdown',
+  checkbox = 'checkbox',
+  custom = 'custom',
+}
+
+export type UserPreferenceDropdownField = {
+  type: UserPreferenceFieldType.dropdown;
+  userSettingsKey: string;
+  defaultValue?: string;
```

Comment:
should this be optional?
What happens if there is no default value provided; will it render an empty selection?

Reply:
IMHO, the default should be optional. If there is no default value provided, then we currently show no selection which means no user preference has been set, which in my opinion is a valid scenario. In such a case we render the placeholder which asks the user to make a selection.